### PR TITLE
ref: Move all consumers to unified consumer CLI

### PR DESCRIPTION
**Pull request by hubertdeng123**
Putting this up again since PR was reverted here: https://github.com/getsentry/self-hosted/pull/2223

Will need to investigate e2e test failures related to profiles topic not being found, resulting in CI being blocked on `getsentry/sentry`

### CODE REVIEW

**Comment by BYK** [comment]
We may need to add some topics here: https://github.com/getsentry/self-hosted/blob/master/install/create-kafka-topics.sh

**Comment by hubertdeng123** [comment]
> We may need to add some topics here: https://github.com/getsentry/self-hosted/blob/master/install/create-kafka-topics.sh

I've double checked, profiles seems to be in that file. It's also the name of the topic used in sentry 🤔
https://github.com/getsentry/sentry/blob/d1c17c650ec94cc85b1277e61b971ee89703b0b2/src/sentry/runner/commands/run.py#L638

Why is this change introducing so much flakiness in our e2e tests...

File: docker-compose.yml [hunk 1]
```diff
@@ -334,19 +334,19 @@ services:
...
-    command: run ingest-consumer --consumer-type=transactions
+    command: run consumer ingest-transactions --consumer-group ingest-consumer
   ingest-replay-recordings:
     <<: *sentry_defaults
-    command: run ingest-replay-recordings
+    command: run consumer ingest-replay-recordings --consumer-group ingest-replay-recordings
   ingest-profiles:
     <<: *sentry_defaults
-    command: run ingest-profiles --no-strict-offset-reset
+    command: run consumer ingest-profiles --consumer-group ingest-profiles --no-strict-offset-reset
```
Comment (BYK) [inline docker-compose.yml:349]: I think this change may fix the issue:
```suggestion
    command: run consumer --no-strict-offset-reset ingest-profiles --consumer-group ingest-profiles
```

Reason being, the new `consumer` command expecting a positional `topic` argument at the end and I suspect putting `--no-strict-offset-reset` at the end might be confusing that.

Comment (hubertdeng123) [reply docker-compose.yml]: thanks for the suggestion, let me try that

**Comment by BYK** [comment]
Sorry this didn't work. Looks like this maybe a bug in Kafka itself: https://github.com/confluentinc/confluent-kafka-dotnet/issues/1766

The solution maybe to just add some retry logic there for this particular error. I think Snuba have something like this already.

**Comment by lynnagara** [comment]
Isn't the issue this line? https://github.com/getsentry/self-hosted/blob/a81cb1e9d9e3407933cfdb1420ee85189f86749b/install/create-kafka-topics.sh#L8

Since there are many topics which contain the substring `profiles` at this point, we entirely skip over creation.

**Comment by BYK** [comment]
@lynnagara that's a good catch. That said I _think_ @hubertdeng123 tried to creat topics without checking at all earlier here: [`55441d0` (#2224)](https://github.com/getsentry/self-hosted/pull/2224/commits/55441d055c03d262f78c8b81e3ed33cd39c71657)

**Comment by untitaker** [comment]
@lynnagara indeed good catch. `-w` flag was supposed to take care of this (i think), but it didn't work for me locally

**Comment by untitaker** [comment]
tests are green 😎

**Comment by untitaker** [comment]
[`55441d0` (#2224)](https://github.com/getsentry/self-hosted/pull/2224/commits/55441d055c03d262f78c8b81e3ed33cd39c71657) would've almost worked as well if we added `|| true` to catch the error when the topic already exists. why the topic would already exist I don't know

**Comment by lynnagara** [comment]
> looking at the CI output it seems that [`55441d0` (#2224)](https://github.com/getsentry/self-hosted/pull/2224/commits/55441d055c03d262f78c8b81e3ed33cd39c71657) would've almost worked as well if we added `|| true` to catch the error when the topic already exists. why the topic would already exist I don't know

A few topics exist b/c `snuba bootstrap` runs before this and creates them

**Review by lynnagara — APPROVED** [review]

**Comment by untitaker** [comment]
> A few topics exist b/c snuba bootstrap runs before this and creates them

should we remove it? having two ways to create topics sounds sketchy. I don't know if we can make it so that sentry creates some of the topics and snuba some of the others, it's unclear to me which service "owns" a particular topic

**Comment by lynnagara** [comment]
> > A few topics exist b/c snuba bootstrap runs before this and creates them
> 
> should we remove it? having two ways to create topics sounds sketchy. I don't know if we can make it so that sentry creates some of the topics and snuba some of the others, it's unclear to me which service "owns" a particular topic

The thing is right now Snuba actually does create them with the correct configs, so it's a bit better to create those ones there. We should move the source of truth for all topics and topic configuration into `sentry-kafka-schemas` and avoid all this?

**Comment by BYK** [comment]
Wow, great catch @lynnagara and @untitaker. I think I'm responsible for that fuzzy topic matching so I apologize for the pain I caused 😅 

> We should move the source of truth for all topics and topic configuration into sentry-kafka-schemas and avoid all this?

I present you another ancient issue then: #857 😀

**Pull request merged**